### PR TITLE
move all logic for configuring remote files to client and move sc-con…

### PR DESCRIPTION
**Pull request by gadfort**
…figure to sc-remote

### CODE REVIEW

File: docs/development_guide/remote_processing.rst [hunk 1]
```diff
@@ -38,7 +38,7 @@ If a previous credentials file already exists, you will be prompted to overwrite
 Private Server
 ^^^^^^^^^^^^^^
 
-If your custom remote server requires authentication, you can run ``sc-configure`` with no arguments and fill in the address, username, and password fields that it prompts you for.
+If your custom remote server requires authentication, you can run ``sc-remote -configure`` with additional no arguments and fill in the address, username, and password fields that it prompts you for.
```
Comment (troibe) [inline docs/development_guide/remote_processing.rst:41]: You probably didn't mean to put `additional` there.

File: docs/development_guide/remote_processing.rst [hunk 2]
```diff
@@ -50,14 +50,14 @@ SiliconCompiler also supports private servers which require authentication to ac
...
 .. code-block:: console
 
-  (venv) sc-configure
-  Remote server address: your-server
-  Remote username: your-username
-  Remote password: your-key
+  (venv) sc-remote -configure
+  Remote server address (leave blank to use default server): your-server
+  Remote username (leave blank for no username): your-username
+  Remote password (leave blank for no password): your-key
```
Comment (troibe) [inline docs/development_guide/remote_processing.rst:60]: Setting the port would have been very useful for my usecase.

File: tests/apps/test_sc_remote.py [hunk 3]
```diff
@@ -65,7 +66,6 @@ def versions():
 
 ###########################
 @pytest.mark.quick
-@pytest.mark.timeout(30)
```
Comment (troibe) [inline tests/apps/test_sc_remote.py:68]: Was this causing issues before?

File: docs/development_guide/remote_processing.rst (hunk 2 above)
Comment (gadfort) [reply docs/development_guide/remote_processing.rst:60]: you can specify the port via `your-server:your-port`

File: tests/apps/test_sc_remote.py (hunk 3 above)
Comment (gadfort) [reply tests/apps/test_sc_remote.py:68]: no, the default timeout is 120s, no reason to make it shorter

File: siliconcompiler/apps/sc_remote.py [hunk 4]
```diff
@@ -29,31 +30,50 @@ def main():
     chip = Chip(progname)
     switchlist = ['-cfg', '-credentials']
     extra_args = {
-        '-reconnect': {'action': 'store_true', 'required': False},
-        '-cancel': {'action': 'store_true', 'required': False},
-        '-delete': {'action': 'store_true', 'required': False},
+        '-configure': {'action': 'store_true',
+                       'help': 'create configuration file for the remote'},
+        '-server': {'help': 'address of server for configure',
```
[the remark this answers is not in the document]
Comment (gadfort) [reply siliconcompiler/apps/sc_remote.py:49]: fixed the help string to make it clear how to set it

**Review by troibe — APPROVED** [review]
LGTM

**Pull request merged**